source/OptimaJet.DWKit.StarterApplication/Services/HttpCurrentUserContext.cs

```diff
@@ -26,5 +68,69 @@ public class HttpCurrentUserContext : ICurrentUserContext
             }
         }
 
+        public string Email {
+            get {
+                if (email == null) {
+                    email = this.HttpContext.GetAuth0Email();
+                }
+                return email;
+            }
+        }
+
+        public string GivenName {
+            get {
+                if (givenName == null) {
+                    var auth = AuthType;
+                    if (string.Compare(auth, "auth0", StringComparison.Ordinal) == 0) {
+                        try
+                        {
+                            // Use Auth0 Management API to get value
+                            givenName = Auth0User.UserMetadata.given_name;
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.Error(ex, $"Failed to request given_name from Auth0: auth0id={Auth0Id}");
+                        }
+                    } else {
+                        givenName = this.HttpContext.GetAuth0GivenName();
+                    }
+                }
+                return givenName;
+            }
+        }
+
+        public string FamilyName {
+            get {
+                if (familyName == null) {
+                    var auth = AuthType;
+                    if (string.Compare(auth, "auth0", StringComparison.Ordinal) == 0)
+                    {
+                        try
```

Comment:
I don't think this is the best use of try/catch here.
functionality wise, it doesn't get us anything additional to not having it (other than adding to the logs, which I think is more info than we need)

Also, what Exception would be thrown?

Catching the root `Exception` class can accidentally catch unexpected errors where you would usually want those to bubble up.

Reply:
If we catch it here, then the GivenName and FamilyName are not assign which is not a big deal.  This is just a convenience and not critical (since the user can enter these in their profile).  If the exception goes out of here it could cause the account creation to fail.  I would prefer to swallow exceptions here.